frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/ArchiveExtractor.java

```diff
@@ -29,86 +34,101 @@
 }
 
 final class DefaultArchiveExtractor implements ArchiveExtractor {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DefaultArchiveExtractor.class);
+
     private void prepDestination(File path, boolean directory) throws IOException {
         if (directory) {
             path.mkdirs();
         } else {
             if (!path.getParentFile().exists()) {
                 path.getParentFile().mkdirs();
             }
-            if(!path.getParentFile().canWrite()) {
-                throw new AccessDeniedException(
-                        String.format("Could not get write permissions for '%s'", path.getParentFile().getAbsolutePath()));
+            if (!path.getParentFile().canWrite()) {
+                throw new AccessDeniedException(String.format("Could not get write permissions for '%s'",
+                    path.getParentFile().getAbsolutePath()));
             }
         }
     }
 
     @Override
     public void extract(String archive, String destinationDirectory) throws ArchiveExtractionException {
         final File archiveFile = new File(archive);
-        FileInputStream fis = null;
-        try {
-            fis = new FileInputStream(archiveFile);
 
-            if("zip".equals(FileUtils.getExtension(archiveFile.getAbsolutePath()))){
+        try (FileInputStream fis = new FileInputStream(archiveFile)) {
+            if ("msi".equals(FileUtils.getExtension(archiveFile.getAbsolutePath()))) {
+                String command = "msiexec /a " + archiveFile.getAbsolutePath() + " /qn TARGETDIR=\""
```

Comment:
Will this cause the yarn installation to be registered globally in the system, even if the files are placed in the local folder? The operating system shouldn't know about yarn being installed (because then it can be accidentally uninstalled).